gym_ignition/robots/base/pybullet_robot.py

```diff
@@ -330,20 +340,28 @@ def set_joint_control_mode(self, joint_name: str, mode: JointControlMode) -> boo
 
     def joint_position(self, joint_name: str) -> float:
         joint_idx = self._joints_name2index[joint_name]
-        return self._pybullet.getJointState(self._robot_id, joint_idx)[0]
+        state = JointStatePyBullet._make(
+            self._pybullet.getJointState(self._robot_id, joint_idx))
+        return state.jointPosition
```

Comment:
Pybullet API in my opinion are a bit messy under this point of view. In this particular case (but there are a lot of them as you can read in its documentation) the `getJointState` method returns a tuple with many fields containing the joint data. You can access an individual field in a tuple using an index (`[0]` in the previous implementation). In order to improve the readability of the code, I introduced a `NamedTuple` object defined in the same file and I cast the output of the `getJointState` method to that class. In this way, I can access the fields of the tuple with their name instead of a arcane index (`jointPosition` in this case).

Reply:
Great, I guess it's more interpretable and maintainable in this way.